tsfresh/utilities/dataframe_functions.py

```diff
@@ -372,7 +375,55 @@ def _normalize_input_to_internal_representation(timeseries_container, column_id,
     return timeseries_container, column_id, column_kind, column_value
 
 
-def roll_time_series(df_or_dict, column_id, column_sort, column_kind, rolling_direction, max_timeshift=None):
+def _roll_out_time_series(time_shift, grouped_data, rolling_direction, max_timeshift, min_timeshift,
+                          column_sort, column_id):
+    """
+    Internal helper function for roll_time_series.
+    This function has the task to extract the rolled forecast data frame of the number `time_shift`.
+    This means it is `time_shift` in the future - starting counting from the first row
+    for each id (and kind).
+    This means we cut out the data until `time_shift`.
+    The first row we cut out is either 0 or given by the maximal allowed length of `max_timeshift`.
+    for a negative rolling direction it is reversed
+    """
+    def _f(x):
```

Comment:
I would expose `time_shift`, `max_timeshift`, etc. as parameters of _f, makes it also easier to test that method.

could rename `time_shift` to start here as well

Reply:
The reason I did not expose them, is that I can then just call `groupby.apply(_f)` and not
`groupby.apply(lambda x: _f(x, param, param, param)`.

Please note, that the `time_shift` parameter is not the start point!